src/include/settings/settings_manager.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <gflags/gflags.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include "catalog/settings_handle.h"
+#include "common/exception.h"
+#include "loggers/settings_logger.h"
+#include "main/main_database.h"
+#include "settings/settings_param.h"
+#include "type/transient_value.h"
+#include "type/transient_value_peeker.h"
+
+#define __SETTING_GFLAGS_DECLARE__     // NOLINT
+#include "settings/settings_common.h"  // NOLINT
+#include "settings/settings_defs.h"    // NOLINT
+#undef __SETTING_GFLAGS_DECLARE__      // NOLINT
+
+namespace terrier::settings {
+
+using callback_fn = void (*)(void *, void *);
+
+/**
+ * A wrapper for pg_settings table, does not store values in it.
+ * Stores and triggers callbacks when a tunable parameter is changed.
+ * Static class.
+ */
+
+class SettingsManager {
+ public:
+  SettingsManager() = delete;
+  SettingsManager(const SettingsManager &) = delete;
+
+  /**
+   * The constructor of settings manager
+   * @param catalog a shared pointer to the system catalog
+   * @param txn_manager a pointer to the transaction manager
+   */
+  SettingsManager(const std::shared_ptr<catalog::Catalog> &catalog, transaction::TransactionManager *txn_manager);
```

Comment:
Not understand why use `shared_ptr<> &`. I think a reference is better because `shared_ptr<> &` may affect the performance